phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/PhoenixConfigurationUtil.java

```diff
@@ -192,6 +196,18 @@ public static void setOutputTableName(final Configuration configuration, final S
     public static void setUpsertColumnNames(final Configuration configuration,final String[] columns) {
         setValues(configuration, columns, MAPREDUCE_UPSERT_COLUMN_COUNT, MAPREDUCE_UPSERT_COLUMN_VALUE_PREFIX);
     }
+
+    public static void setSnapshotNameKey(final Configuration configuration, final String snapshotName) {
+        Preconditions.checkNotNull(configuration);
+        Preconditions.checkNotNull(snapshotName);
+        configuration.set(SNAPSHOT_NAME_KEY, snapshotName);
```

Comment:
The idea of having the snapshot name in the configuration is not going to translate well when we want to expose snapshot reads for queries in general as the configuration is a global object.